torch/nn/functional.py

```diff
@@ -821,6 +822,55 @@ def upsample_bilinear(input, size=None, scale_factor=None):
     return upsample(input, size, scale_factor, mode='bilinear')
 
 
+def grid_sample(input, grid, mode='bilinear'):
+    """Given an :attr:`input` and a flow-field :attr:`grid`, computes the
+    `output` using input pixel locations from the grid.
+
+    Uses bilinear interpolation to sample the input pixels.
+    Currently, only spatial (4 dimensional) inputs are supported.
+
+    For each output location, :attr:`grid` has `x` and `y`
+    input pixel locations which are used to compute output.
+
+    :attr:`grid` has values in the range of `[-1, 1]`. This is because the
+    pixel locations are normalized by the input height and width.
+
+    For example, values: x: -1, y: -1 is the left-top pixel of the input
+                 values: x: 1, y: 1 is the right-bottom pixel of the input
+
+    If :attr:`grid` has values outside the range of `[-1, 1]`, those locations
+    are ignored (i.e. 0 is used as a contribution to the bilinear interpolation)
+
+    .. Note:: This function is used in building Spatial Transformer Networks
+
+    Args:
+        input (Variable): input batch of images (N x C x H x W)
+        grid (Variable): flow-field of size (N x H x W x 2)
+
+    Returns:
+        output (Variable): output Tensor
+
+    """
+    batch_size, channels, in_height, in_width = input.size()
+    return GridSampler.apply(input, grid)
+
+
+def affine_grid(theta, size):
+    """Generates a 2d flow field, given a batch of affine matrices :attr:`theta`
+    Generally used in conjunction with :function:`grid_sample` to
+    implement Spatial Transformer Networks.
+
+    Args:
+        theta (Variable): input batch of affine matrices (N x 2 x 3)
+        size (torch.Size): the target output image size (N x H x W x 2)
```

Comment:
The C parameter is used in cudnn's SpatialTransformerDescriptor. It's not needed (for example, CPU doesn't use it), but that's how CuDNN API seems to be.

I'll fix the confusing wording.